### ISSUE TITLE
Camera stream not visible for other participants when using Chrome

### ISSUE THREAD

**Issue by Juppstein** [bekriebel/fvtt-module-avclient-livekit#66]
I am using Google Chrome (Version 124.0.6367.60 (Official Build) (64-bit)) on Ubuntu 22.04 and I have the issue that my camera stream is not visible to other participants, while they can see and hear each other without an issue. The problem goes away when I switch from Chrome to Firefox (125.0 64-bit). 

My LiveKit system is selfhosted on a headless Ubuntu 22.04 server. The reverse proxy is nginx with the auto updating ssl certs using certbox, basically the setup you describe in your wiki entry. 

Are there any settings Chrome that can block video stream transmissions? I found it a bit odd that while Chrome does not work, Firefox worked right away. If there are logs that I could upload I'd need a tip where to fetch them from.

Cheers!

**Comment by ashmedai711** [comment]
Are you getting any errors in the console (F12)? I'm having a similar issue in chrome 124.0.6367.61 and getting this error.
<img width="643" alt="image" src="https://github.com/bekriebel/fvtt-module-avclient-livekit/assets/119912908/3dc44931-2f19-40ba-b0f5-769a94b9c5da">
Same issue in Edge 124.0.2478.51, but no issue in Firefox 125.0 nor in Foundry Lightweight Client v4.3.1. I'm using Molten Hosting's livekit server.

**Comment by Juppstein** [comment]
Yup that's the one. Same error.

**Comment by bekriebel** [comment]
@ashmedai711, are you on Linux as well? Is this Google Chrome, or Chromium? It looks like there's a codec issue. I wonder if there was a recent change that is not allowing the default codecs to work. I'll have to test this in my dev environment next week to see what I need to change.

**Comment by ashmedai711** [comment]
@bekriebel I'm on Windows 11 and this is Google Chrome, but after doing some more digging it, it looks like perhaps [there is an issue](https://issues.chromium.org/issues/331565934?pli=1) in the Chromium source code of version 124 that is waiting to be fixed? I'm not that knowledgeable, but that might explain why Edge shows the same issue and not Firefox.

**Comment by bekriebel** [comment]
Good catch, that may be it.

In the meantime, I believe you can set the preferred codec at the server level. You may want to set your servers to VP8 or VP9 until that is resolved.

I'll test it out next week either way.

**Comment by Juppstein** [comment]
Sorry for the question, but where would I configure the room codecs for a system running livekit in docker/caddy setup?

**Comment by ashmedai711** [comment]
> Sorry for the question, but where would I configure the room codecs for a system running livekit in docker/caddy setup?

I'm not sure myself since I don't own the server I use, but I believe I was able to do a temporary fix by editing the module code in avclient-livekit.bundle.js to add `if (e.mimeType === 'video/AV1') return;` in this section. Probably not the ideal solution, so use at your own risk, but it seems to be working now for me.
```js 
setPreferredCodec(e, t, i) {
                if (!("getCapabilities"in RTCRtpSender))
                    return;
                const n = RTCRtpSender.getCapabilities(t);
                if (!n)
                    return;
                u.debug("get capabilities", n);
                const s = []
                  , r = []
                  , o = [];
                n.codecs.forEach((e=>{
                    if (e.mimeType === 'video/AV1') return;
                    const t = e.mimeType.toLowerCase();
                    if ("audio/opus" === t)
                        return void s.push(e);
                    t === "video/".concat(i) ? "h264" !== i || e.sdpFmtpLine && e.sdpFmtpLine.includes("profile-level-id=42e01f") ? s.push(e) : r.push(e) : o.push(e)
                }

**Comment by bekriebel** [comment]
I've pushed a possible fix for this in version 0.5.26. Can you give it a shot and see if it fixes the issue for you?

**Comment by ashmedai711** [comment]
@bekriebel Yes, that works for me! Thanks!
